### DISCUSSION TITLE
Could I create a temporary file for a workflow instance

### DISCUSSION THREAD

**Discussion by zhutinglei** [apache/dolphinscheduler#7118] [Q&A] [answered]
I would like to create a temporary file for a workflow instance, which means
- There will be an individual directory for each workflow instance
- I can create a file, such as `a.txt` in the first task node
- I can read the file in the second task node

I tried to define a shell task with the cmd `touch a.txt`, but I got a `Permission Denied` error when I execute the task.

It is similar to passing variables between task nodes, but there are differences for us. The existing ancestral software can only deal with formatted text files, but our novel data management and reduction system will be constructed based on a database.  Hence, I have to generate temporary files first and call the softwares.

**Comment by zhongjiajie** [comment, marked as answer]
Hi @zhutinglei, when you ask question, English first for more general

**Reply by zhutinglei** [reply]
Changed

**Reply by zhongjiajie** [reply]
Thanks @zhutinglei  for changing. DS run shell command with specific user in linux, it call `tenant` in DS space, and it is a attribute in DS's user, So you could go and find whether you `tenant` have permission to create folder in your worker. Hope it could help you.

**Reply by zhongjiajie** [reply]
BTW, please also change you discussion title, thx.

**Reply by zhutinglei** [reply]
Thank you for your explanation. Now I created a user (tlzhu)  in linux first and then link the DS user to this tenant (tlzhu), then I can create folder and file in `/home/tlzhu` directory. 

But what I really want to do is to let DS user can create folder and file in the task instance folder, such as 
```
t/tmp/dolphinscheduler/exec/process/930444347523072/970585450299392_1
```

I use `chmod 777 -R /tmp` for a test, and then I find that the task instance folder is created by root, user tlzhu have no write and read permission.
```
tlzhu@996287f4e23b:/tmp/dolphinscheduler/exec/process/930444347523072/970585450299392_1$ ls -lah
total 24K
drwxrwxrwx  6 root root 4.0K Dec  9 20:13 .
drwxrwxrwx 13 root root 4.0K Dec  9 19:45 ..
drwxrwxrwx  2 root root 4.0K Dec  9 20:02 45
drwxr-xr-x  2 root root 4.0K Dec  9 20:03 46
drwxr-xr-x  2 root root 4.0K Dec  9 20:07 47
drwxr-xr-x  2 root root 4.0K Dec  9 20:13 48
```

I guess it is not the best practice to let task node create/modify files in the task instance folder, because it may also let the tenant be able to modify files in others' task instance folders. Do you have any suggestions on this?
